### PR TITLE
Programmes crud api

### PR DESCRIPTION
**Pull request by abdurrahmanfaqihiskandar**
Added a schema for programmes.

Added CRUD endpoints for programmes.

Still working on getting an organisation's programmes.

### CODE REVIEW

File: src/controllers/programmes.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ * @access  Owner
+ */
+export const updateProgramme = asyncHandler(async (req, res, next) => {
+    // check if programme exists
+    const isValidProgramme = await db.oneOrNone('SELECT * FROM programmes WHERE programme_id = $1', req.params.id);
+
+    // return bad request response if invalid organisation
+    if (!isValidProgramme) {
+        return next(new ErrorResponse(`Programme does not exist`, 400));
+    }
```
Comment (don-tay) [inline src/controllers/programmes.ts:80]:
```suggestion
    // check if programme exists
await  db.one('SELECT * FROM programmes WHERE programme_id = $1', req.params.id);
```

File: src/controllers/programmes.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * @access  Owner
+ */
+export const deleteProgramme = asyncHandler(async (req, res, next) => {
+    // check if programme exists
+    const programme = await db.one('SELECT * FROM programmes WHERE programme_id = $1', req.params.id);
+
+    // Unauthorised if not owner
+    if (req.user.role !== 'admin') {
+        return next(new ErrorResponse(`User not authorised to delete this programme`, 403));
+    }
```
Comment (don-tay) [inline src/controllers/programmes.ts:115]: Must add check to allow for organisation owner

File: src/routes/api/organisation-routes.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// Re-route this URI to other resource router
+router.use('/:organisation_id/programmes', programmeRoute);
+
+// Define input validation
+const validateCreateOrganisationFields = [
+    check('name', INVALID_ALPHA_SPACE_MSG('name')).trim().notEmpty(),
+    check('type', INVALID_ALPHA_SPACE_MSG('type')).optional().trim(),
+    check('about', INVALID_ALPHA_SPACE_MSG('about')).trim().notEmpty(),
+    check('website_url', INVALID_URL_MSG).optional().trim().isURL(),
+    check('handphone', INVALID_PHONE_NUMBER_MSG).trim().notEmpty(),
```
Comment (don-tay) [inline src/routes/api/organisation-routes.ts:22]: Let's make handphone optional for now

**Review by don-tay — CHANGES_REQUESTED** [review]
The database schema looks good. For the endpoints, we can be more lenient in our data input requirements.

Lets review the organisations and programmes creation flow with FE after this PR, so that we can work on the permissions and `is_verirfied` flag

File: src/database/schema.sql [hunk 4]
```diff
@@ -123,11 +123,12 @@ CREATE TABLE ProfileSkills (
 CREATE TABLE Organisations (
 	organisation_id SERIAL,
 	name VARCHAR NOT NULL,
-	TYPE VARCHAR NOT NULL,
+	TYPE VARCHAR,
```
Comment (don-tay) [inline src/database/schema.sql:126]:
```suggestion
	type VARCHAR,
```

File: src/database/schema.sql [hunk 5]
```diff
@@ -123,11 +123,12 @@ CREATE TABLE ProfileSkills (
 CREATE TABLE Organisations (
```
Comment (don-tay) [inline src/database/schema.sql:123]: For `Organisations`, additionally add 
`locations VARCHAR`, optional field
`story TEXT` optional field

Comment (don-tay) [reply src/database/schema.sql:123]: Do add these columns on the create and update endpoints

**Review by don-tay — APPROVED** [review]
Small updates to the schema (and change the endpoint naming to fit this as well) and it LGTM

**Review by Kalsyc — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**